File: bakamc-common/src/main/kotlin/cn/bakamc/common/config/Configs.kt
package cn.bakamc.common.config

import moe.forpleuvoir.nebula.common.util.plus
import moe.forpleuvoir.nebula.common.util.second
import moe.forpleuvoir.nebula.config.category.ConfigCategoryImpl
import moe.forpleuvoir.nebula.config.manager.AutoSaveConfigManager
import moe.forpleuvoir.nebula.config.manager.LocalConfigManager
import moe.forpleuvoir.nebula.config.persistence.JsonConfigManagerPersistence
import java.nio.file.Path
import java.util.*

object Configs : LocalConfigManager("bakamc"), AutoSaveConfigManager, JsonConfigManagerPersistence {

    override lateinit var configPath: Path
        private set

    override val starTime: Date = Date() + 30.second
    override val period: Long = 30.second

    override fun init() {
        super<LocalConfigManager>.init()
        super<AutoSaveConfigManager>.init()
    }

    /**
     * 配置文件路径
     * @param path Path
     */
    fun init(path: Path) {
        configPath = path
        init()
    }

    object FlightEnergy: ConfigCategoryImpl("flight_energy"){




    }

}

File: bakamc-folia/src/main/scala/cn/bakamc/folia/config/Configs.scala
package cn.bakamc.folia.config

import moe.forpleuvoir.nebula.config.Comment

object Configs extends PluginConfigManager("config") {

  @Comment("杂项")
  val misc: MiscConfig.type = MiscConfig


  @Comment("飞行能量")
  val flightEnergy: FlightEnergyConfig.type = FlightEnergyConfig

  @Comment("实体相关配置")
  val entity: EntityConfig.type = EntityConfig



}
